core-reactive/src/main/java/io/micronaut/core/async/publisher/Publishers.java

```diff
@@ -298,8 +298,8 @@ protected void doOnSubscribe(Subscription subscription) {
             @Override
             protected void doOnNext(T message) {
                 try {
-                    actual.onNext(message);
                     consumer.accept(message);
+                    actual.onNext(message);
```

Comment:
This is debatable whether it should be done. The `then` in the method implies to me it happens after the `onNext` but the javadoc isn't clear